beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1299,9 +1305,140 @@ impl<T: BeaconChainTypes> ExecutionPendingBlock<T> {
+                BlockError::ExecutionPayloadError(ExecutionPayloadError::RequestFailed(e))
+            })?;
+            let num_fetched_tx = response.blobs.iter().filter(|b| b.is_some()).count();
+            if num_fetched_tx == 0 {
+                debug!(chain.log, "Blobs from EL - response with none");
+                return Ok(());
+            } else if num_fetched_tx < num_blob_tx {
+                debug!(
+                    chain.log,
+                    "Blobs from EL - response with some";
+                    "fetched" => num_fetched_tx,
+                    "total" => num_blob_tx,
+                );
+            } else {
+                debug!(
+                    chain.log,
+                    "Blobs from EL - response with all";
+                    "num_blob_tx" => num_fetched_tx
+                );
+            }
+            let (signed_block_header, kzg_commitments_proof) =
+                block.signed_block_header_and_kzg_commitments_proof()?;
+
+            let mut fixed_blob_sidecar_list = FixedBlobSidecarList::default();
+            for (i, (blob, kzg_proof)) in blob_start_indices
+                .into_iter()
+                .zip(response.blobs)
+                .filter_map(|(start_index, blobs)| blobs.map(|blobs| (start_index, blobs)))
+                .flat_map(|(start_index, blob)| {
+                    blob.blobs
+                        .into_iter()
+                        .zip(blob.proofs)
+                        .enumerate()
+                        .map(move |(i, v)| (start_index + i, v))
+                })
+            {
+                match BlobSidecar::new_efficiently(
+                    i,
+                    blob,
+                    &block,
+                    signed_block_header.clone(),
+                    &kzg_commitments_proof,
+                    kzg_proof,
+                ) {
+                    Ok(blob) => {
+                        if let Some(blob_mut) = fixed_blob_sidecar_list.get_mut(i) {
+                            *blob_mut = Some(Arc::new(blob));
+                        } else {
+                            error!(
+                                chain.log,
+                                "Blobs from EL - out of bounds";
+                                "i" => i
+                            );
+                        }
+                    }
+                    Err(e) => {
+                        warn!(
+                            chain.log,
+                            "Blobs from EL - error";
+                            "error" => ?e
+                        );
+                    }
+                }
+            }
+            debug!(
+                chain.log,
+                "Blobs from EL - start processing";
+                "num_blobs" => fixed_blob_sidecar_list.iter().filter(|b| b.is_some()).count(),
+            );
+            chain
+                .process_rpc_blobs(block.slot(), block_root, fixed_blob_sidecar_list)
```

Comment:
Oops, should update this to `process_engine_blobs` to skip the KZG proof verification